### PR TITLE
Parse more operators in requirements

### LINKED ISSUE #331: requirements-txt-fixer does not parse all requirements
While working on #330 I noticed that the requirements-txt-fixer hook only parses a small subset of requirement specifiers. Take this example:

    pyramid-foo==1
    pyramid>=2

`pyramid` should sort before `pyramid-foo`, but the hook interprets the `>=2` as part of the name.

According to [PEP 508](https://www.python.org/dev/peps/pep-0508/) there are a couple of things that can be specified after the name (extras, versions, markers) so maybe an approach would be to use a regular expression to extract only the package name?

### PR DESCRIPTION
**Pull request by mxr**
Resolves #331

### CODE REVIEW

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 1]
```diff
@@ -10,18 +11,31 @@
 
 
 class Requirement:
+    UNTIL_COMPARISON = re.compile(b'={2,3}|!=|~=|>=?|<=?')
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py:14]: is this supposed to match `===` ?

File: pre_commit_hooks/requirements_txt_fixer.py [hunk 2]
```diff
@@ -10,18 +11,31 @@
...
+    UNTIL_SEP = re.compile(rb'\S[^;\s]+')
```
Comment (asottile) [inline pre_commit_hooks/requirements_txt_fixer.py:15]: is this different from `[^;\s]+`?

File: pre_commit_hooks/requirements_txt_fixer.py (hunk 1 above)
Comment (mxr) [reply pre_commit_hooks/requirements_txt_fixer.py:14]: Yes, we match it here too https://github.com/asottile/setup-cfg-fmt/blob/c3a291b4615a1ed07b275dd37d20df9ca02f8880/setup_cfg_fmt.py#L228

File: pre_commit_hooks/requirements_txt_fixer.py (hunk 2 above)
Comment (mxr) [reply pre_commit_hooks/requirements_txt_fixer.py]: Good call, I got confused parsing the `--hash` lines

File: pre_commit_hooks/requirements_txt_fixer.py (hunk 1 above)
Comment (asottile) [reply pre_commit_hooks/requirements_txt_fixer.py:14]: huh, TIL I guess

**Pull request merged**